### DISCUSSION TITLE
Packaging BiblioteQ in other Linux formats.

### DISCUSSION THREAD

**Discussion by Nerdy-boi** [textbrowser/biblioteq#250] [Ideas]
The project would be much better in my opinion of the developers would publish either a tarball, or packacge in either flatpak or appimage for us that dont run debian based linux systems.

**Comment by textbrowser** [comment]
Which developers?

**Comment by textbrowser** [comment]
The releases include source files. I could create a bundle with the binary without the libraries or with the libraries but that's a lot of work because I'd have to include every single dependency. This would not guarantee that the executable would work on your system.

**Comment by textbrowser** [comment]
I grow tired of creating release files for people. If you're interested in BQ, you should spend your time building it and / or request that your distribution adopt it. :) The expectation that volunteers of free software should also volunteer their time to create files for every distribution is absurd. I don't have a factory of release engineers.

**Comment by FriedrichFroebel** [comment]
Source packages are already available as tarballs, created automatically by GitHub on each release.

Flatpak or AppImage packages are distribution-independent as far as I am aware, as they bundle the required runtime as well. As far as I am aware, lots of other projects have release automations to automatically build the desired releases through continuous integration (CI), with GitHub already providing such support through its own GitHub Actions. Once setup, it should (at least in theory) not require much maintenance. This would avoid the current manual build process for at least the most common targets (and if I remember correctly, this has been discussed briefly in another issue). And yes, another alternative would be to maintain actual system packages (through a PPA (usually pre-built packages) or through the AUR package index (usually source-based packages) for example), which does not necessarily has to be done by you as the actual developer, but by interested users/maintainers.

**Comment by textbrowser** [comment]
Should be a ticket instead of a discussion.

**Comment by textbrowser** [comment]
For a single library, libyaz.

objdump -p ./BiblioteQ | grep NEEDED
  NEEDED               libyaz.so.5
  NEEDED               libQt5PrintSupport.so.5
  NEEDED               libQt5Widgets.so.5
  NEEDED               libQt5Gui.so.5
  NEEDED               libQt5Network.so.5
  NEEDED               libQt5Sql.so.5
  NEEDED               libQt5Core.so.5
  NEEDED               libstdc++.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
objdump -p /usr/lib/x86_64-linux-gnu/libyaz.so.5 | grep NEEDED
  NEEDED               libgnutls.so.30
  NEEDED               libwrap.so.0
  NEEDED               libexslt.so.0
  NEEDED               libxslt.so.1
  NEEDED               libxml2.so.2
  NEEDED               libc.so.6
objdump -p /usr/lib/x86_64-linux-gnu/libgnutls.so.30 | grep NEEDED
  NEEDED               libp11-kit.so.0
  NEEDED               libidn2.so.0
  NEEDED               libunistring.so.2
  NEEDED               libtasn1.so.6
  NEEDED               libnettle.so.8
  NEEDED               libhogweed.so.6
  NEEDED               libgmp.so.10
  NEEDED               libc.so.6
  NEEDED               ld-linux-x86-64.so.2

This request doesn't have a simple solution.

**Comment by textbrowser** [comment]
ldd ./BiblioteQ 
        linux-vdso.so.1 (0x00007ffc1e1f6000)
        libyaz.so.5 => /lib/x86_64-linux-gnu/libyaz.so.5 (0x00007f67bfe48000)
        libQt5PrintSupport.so.5 => /lib/x86_64-linux-gnu/libQt5PrintSupport.so.5 (0x00007f67bfdd0000)
        libQt5Widgets.so.5 => /lib/x86_64-linux-gnu/libQt5Widgets.so.5 (0x00007f67bf70f000)
        libQt5Gui.so.5 => /lib/x86_64-linux-gnu/libQt5Gui.so.5 (0x00007f67bf031000)
        libQt5Network.so.5 => /lib/x86_64-linux-gnu/libQt5Network.so.5 (0x00007f67bee80000)
        libQt5Sql.so.5 => /lib/x86_64-linux-gnu/libQt5Sql.so.5 (0x00007f67bee3b000)
        libQt5Core.so.5 => /lib/x86_64-linux-gnu/libQt5Core.so.5 (0x00007f67be8dc000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f67be6b2000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f67be692000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f67be46a000)
        libgnutls.so.30 => /lib/x86_64-linux-gnu/libgnutls.so.30 (0x00007f67be27f000)
        libwrap.so.0 => /lib/x86_64-linux-gnu/libwrap.so.0 (0x00007f67be271000)
        libexslt.so.0 => /lib/x86_64-linux-gnu/libexslt.so.0 (0x00007f67be258000)
        libxslt.so.1 => /lib/x86_64-linux-gnu/libxslt.so.1 (0x00007f67be216000)
        libxml2.so.2 => /lib/x86_64-linux-gnu/libxml2.so.2 (0x00007f67be034000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f67bdf4d000)
        libGL.so.1 => /lib/x86_64-linux-gnu/libGL.so.1 (0x00007f67bdec6000)
        libpng16.so.16 => /lib/x86_64-linux-gnu/libpng16.so.16 (0x00007f67bde89000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f67bde6d000)
        libharfbuzz.so.0 => /lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007f67bdd9e000)
        libmd4c.so.0 => /lib/x86_64-linux-gnu/libmd4c.so.0 (0x00007f67bdd8c000)
        libgssapi_krb5.so.2 => /lib/x86_64-linux-gnu/libgssapi_krb5.so.2 (0x00007f67bdd38000)
        libdouble-conversion.so.3 => /lib/x86_64-linux-gnu/libdouble-conversion.so.3 (0x00007f67bdd21000)
        libicui18n.so.70 => /lib/x86_64-linux-gnu/libicui18n.so.70 (0x00007f67bd9f2000)
        libicuuc.so.70 => /lib/x86_64-linux-gnu/libicuuc.so.70 (0x00007f67bd7f7000)
        libpcre2-16.so.0 => /lib/x86_64-linux-gnu/libpcre2-16.so.0 (0x00007f67bd76d000)
        libzstd.so.1 => /lib/x86_64-linux-gnu/libzstd.so.1 (0x00007f67bd69e000)
        libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f67bd564000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f67c063a000)
        libp11-kit.so.0 => /lib/x86_64-linux-gnu/libp11-kit.so.0 (0x00007f67bd427000)
        libidn2.so.0 => /lib/x86_64-linux-gnu/libidn2.so.0 (0x00007f67bd406000)
        libunistring.so.2 => /lib/x86_64-linux-gnu/libunistring.so.2 (0x00007f67bd25c000)
        libtasn1.so.6 => /lib/x86_64-linux-gnu/libtasn1.so.6 (0x00007f67bd244000)
        libnettle.so.8 => /lib/x86_64-linux-gnu/libnettle.so.8 (0x00007f67bd1fe000)
        libhogweed.so.6 => /lib/x86_64-linux-gnu/libhogweed.so.6 (0x00007f67bd1b6000)
        libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f67bd132000)
        libnsl.so.2 => /lib/x86_64-linux-gnu/libnsl.so.2 (0x00007f67bd118000)
        libgcrypt.so.20 => /lib/x86_64-linux-gnu/libgcrypt.so.20 (0x00007f67bcfda000)
        liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f67bcfaf000)
        libGLdispatch.so.0 => /lib/x86_64-linux-gnu/libGLdispatch.so.0 (0x00007f67bcef7000)
        libGLX.so.0 => /lib/x86_64-linux-gnu/libGLX.so.0 (0x00007f67bcec1000)
        libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f67bcdf9000)
        libgraphite2.so.3 => /lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007f67bcdd2000)
        libkrb5.so.3 => /lib/x86_64-linux-gnu/libkrb5.so.3 (0x00007f67bcd07000)
        libk5crypto.so.3 => /lib/x86_64-linux-gnu/libk5crypto.so.3 (0x00007f67bccd8000)
        libcom_err.so.2 => /lib/x86_64-linux-gnu/libcom_err.so.2 (0x00007f67bccd2000)
        libkrb5support.so.0 => /lib/x86_64-linux-gnu/libkrb5support.so.0 (0x00007f67bccc2000)
        libicudata.so.70 => /lib/x86_64-linux-gnu/libicudata.so.70 (0x00007f67bb0a4000)
        libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f67bb02e000)
        libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007f67bb021000)
        libtirpc.so.3 => /lib/x86_64-linux-gnu/libtirpc.so.3 (0x00007f67baff3000)
        libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007f67bafcb000)
        libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007f67bae8b000)
        libbrotlidec.so.1 => /lib/x86_64-linux-gnu/libbrotlidec.so.1 (0x00007f67bae7d000)
        libkeyutils.so.1 => /lib/x86_64-linux-gnu/libkeyutils.so.1 (0x00007f67bae76000)
        libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f67bae62000)
        libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f67bae36000)
        libbrotlicommon.so.1 => /lib/x86_64-linux-gnu/libbrotlicommon.so.1 (0x00007f67bae13000)
        libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007f67bae0d000)
        libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f67bae05000)
        libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007f67baded000)
        libmd.so.0 => /lib/x86_64-linux-gnu/libmd.so.0 (0x00007f67badde000)

**Comment by textbrowser** [comment]
Completed.

https://github.com/textbrowser/biblioteq/commit/d4fa3b7255803d09a9395184df52173dbce12a6e
